bigbluebutton-html5/imports/ui/components/notifications-bar/component.jsx

```diff
@@ -16,8 +16,15 @@ const defaultProps = {
 };
 
 export default class NotificationsBar extends Component {
-  constructor(props) {
-    super(props);
+
+  componentDidMount() {
+    // to let the whiteboard know that the presentation area's size has changed
+    window.dispatchEvent(new Event('resize'));
```

Comment:
`ResizeObserver` is an open spec, it is not implemented in any of the browsers yet, unfortunately.

All they have is a dummy listener which checks the size of an element `n` times a second. They call it a "polyfill" for some reason and promise to "switch to a native API when it's released" 😕  
But at this point it looks like it's worse for us, since it will put more pressure on the client, but we don't need to make any changes in 99.9% of the time.

But hey, it looks like they added it to Chrome as an experimental feature a year ago, maybe we can switch to something better in a couple of years 

P.S. what custom layout are you talking about? resizing for the panels?

Frankly speaking I wouldn't worry about it for now, we can always switch to a less efficient option if we don't have a choice. Let's at least try to make the client a little bit lighter.

Reply:
Then at least move it to a HoC so its easier to add/remove and will not pollute the components with code not related to them and will improve code reuse.